src/main/java/com/contrastsecurity/maven/plugin/sdkx/Scan.java

```diff
@@ -5,17 +5,183 @@
 /** TODO[JG] JAVA-3298 move this to the Contrast Java SDK and flesh it out */
 public final class Scan {
 
+  /**
+   * static factory that enforces invariants for a waiting scan
+   *
+   * @param id unique ID of this scan
+   * @param projectId unique ID of the Scan project
+   * @param organizationId unique Contrast organization ID
+   * @return new waiting {@code Scan}
+   */
+  public static Scan createWaiting(
+      final String id, final String projectId, final String organizationId) {
+    return new Scan(id, projectId, organizationId, Status.WAITING, null);
+  }
+
+  /**
+   * static factory that enforces invariants for a running scan
+   *
+   * @param id unique ID of this scan
+   * @param projectId unique ID of the Scan project
+   * @param organizationId unique Contrast organization ID
+   * @return new running {@code Scan}
+   */
+  public static Scan createRunning(
+      final String id, final String projectId, final String organizationId) {
+    return new Scan(id, projectId, organizationId, Status.RUNNING, null);
+  }
+
+  /**
+   * static factory that enforces invariants for a completed scan
+   *
+   * @param id unique ID of this scan
+   * @param projectId unique ID of the Scan project
+   * @param organizationId unique Contrast organization ID
+   * @return new completed {@code Scan}
+   */
+  public static Scan createCompleted(
+      final String id, final String projectId, final String organizationId) {
+    return new Scan(id, projectId, organizationId, Status.COMPLETED, null);
+  }
+
+  /**
+   * static factory that enforces invariants for a canceled scan
+   *
+   * @param id unique ID of this scan
+   * @param projectId unique ID of the Scan project
+   * @param organizationId unique Contrast organization ID
+   * @return new canceled {@code Scan}
+   */
+  public static Scan createCanceled(
+      final String id, final String projectId, final String organizationId) {
+    return new Scan(id, projectId, organizationId, Status.CANCELLED, null);
+  }
```

Comment:
I think the static factory methods and non-static factory methods (for example `toRunning`) can all be package-private and visible for testing. Maybe they should even be moved to a separate utility under test since they really aren't used for production code as far as I can tell.

Reply:
It's true that they're only used for testing, but the tests are not in the same package so I made them public. I wasn't too upset about them being public, because I could see how they would be useful for a user of the SDK in their tests. For example, when the `sdkx` package migrates to the Contrast Java SDK project, then I'd still like access to these factories for creating unit tests in this Maven plug-in project.

These were package-private before I made the `sdkx.scan` package. I'm on the fence about whether or not that package should exist or if everything there should be in the `sdkx` package. Do you have any opinions here?

If we consolidate the `sdkx` and `sdkx.scan` into one package, I could once again make these factories package-private; however, I still feel there's some utility in having them public.

Reply:
It makes sense to me that they’d be useful for testing, but it weirds me out a bit that we’re including what are essentially test utilities in the API. I’d lean towards making a separate utility class for testing, but then I don’t know if we want a completely separate test artifact since it ends up in the API either way. I’d say leave it for now, and consider revisiting when we migrate to the SDK. Presumably we consider all these classes subject to change anyway since this is just a plugin and not a SDK.